src/OmniSharp.Roslyn.CSharp/Services/Refactoring/V2/BaseCodeActionService.cs

```diff
@@ -23,6 +23,8 @@ public abstract class BaseCodeActionService<TRequest, TResponse> : IRequestHandl
     {
         protected readonly OmniSharpWorkspace Workspace;
         protected readonly IEnumerable<ICodeActionProvider> Providers;
+        protected List<CodeFixProvider> OrderedCodeFixProviders;
```

Comment:
You can make this a `Lazy<IEnumerable<CodeFixProvider>` and initialize it like
```
_lazyFixProviders = new Lazy(() => GetOrderedProviders)```

That will let you avoid some of the crazy exception handling below.